pulser/simulation/simulation.py

```diff
@@ -642,19 +642,18 @@ def _build_basis_and_op_matrices(self) -> None:
                 self.basis[proj[0]] * self.basis[proj[1]].dag()
             )
 
-    def _construct_hamiltonian(self, update_and_extract: bool = True) -> None:
+    def _construct_hamiltonian(self, update: bool = True) -> None:
         """Constructs the hamiltonian from the Sequence.
 
         Also builds qutip.Qobjs related to the Sequence if not built already,
         and refreshes potential noise parameters by drawing new at random.
 
         Args:
-            update_and_extract(bool=True): Whether to update the noise
-                parameters and extract the samples from the sequence.
+            update(bool=True): Whether to update the noise parameters.
         """
-        if update_and_extract:
+        if update:
             self._update_noise()
-            self._extract_samples()
+        self._extract_samples()
```

Comment:
This is correct, the initial configs (from SP errors) may have changed and so the extraction needs to take place for every run.